File: my_py_pkg/my_py_pkg/add_two_ints_client_no_oop.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


def main(args=None):
    rclpy.init(args=args)
    node = Node("add_two_ints_client_no_oop")

    client = node.create_client(AddTwoInts, "add_two_ints")
    while not client.wait_for_service(timeout_sec=1.0):
        node.get_logger().warn("service not available, waiting again...")

    request = AddTwoInts.Request()
    request.a = 3
    request.b = 7
    future = client.call_async(request)
    rclpy.spin_until_future_complete(node, future)

    response = future.result()
    if response is None:
        node.get_logger().error("Service call failed")
    else:
        node.get_logger().info(
            f"Result of add_two_ints: a={request.a}, b={request.b}, sum={response.sum}"
        )

    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/add_two_ints_client_no_oop.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using namespace std::chrono_literals;

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("add_two_ints_client_no_oop");

    auto client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
    while (!client->wait_for_service(1s)) {
        RCLCPP_WARN(node->get_logger(), "Service not available, waiting again...");
    }

    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 5;
    request->b = 3;

    auto result_future = client->async_send_request(request);
    if (rclcpp::spin_until_future_complete(node, result_future) ==
        rclcpp::FutureReturnCode::SUCCESS)
    {
        RCLCPP_INFO(node->get_logger(), "Summing %ld + %ld  Result: %ld", request->a, request->b, result_future.get()->sum);
    } else {
        RCLCPP_ERROR(node->get_logger(), "Failed to call service add_two_ints");
    }
    
    rclcpp::shutdown();
    return 0;
}